pkg/cmd/admin/router/router.go

```diff
@@ -469,26 +507,29 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			generate = true
 		}
 	}
+	if !generate {
+		fmt.Fprintf(out, "Router %q service exists\n", name)
+		return nil
+	}
 
-	if generate {
-		if cfg.DryRun && !output {
-			return fmt.Errorf("router %q does not exist (no service)", name)
-		}
-
-		if len(cfg.ServiceAccount) == 0 {
-			return fmt.Errorf("router could not be created; you must specify a service account with --service-account")
-		}
+	if cfg.DryRun && !output {
+		return fmt.Errorf("router %q does not exist (no service)", name)
+	}
 
-		err := validateServiceAccount(kClient, namespace, cfg.ServiceAccount)
-		if err != nil {
-			return fmt.Errorf("router could not be created; %v", err)
-		}
+	if len(cfg.ServiceAccount) == 0 {
+		return fmt.Errorf("you must specify a service account for the router with --service-account")
+	}
 
-		// create new router
-		if len(cfg.Credentials) == 0 {
-			return fmt.Errorf("router could not be created; you must specify a .kubeconfig file path containing credentials for connecting the router to the master with --credentials")
-		}
+	if err := validateServiceAccount(kClient, namespace, cfg.ServiceAccount, cfg.HostNetwork); err != nil {
+		return fmt.Errorf("router could not be created; %v", err)
```

Comment:
This was intentional, the error may not be related to the failure

On Wed, Feb 17, 2016 at 12:14 PM, Michail Kargakis <notifications@github.com
> wrote:

> In pkg/cmd/admin/router/router.go
> <https://github.com/openshift/origin/pull/7251#discussion_r53197443>:
>
> >
> > -		// create new router
> > -		if len(cfg.Credentials) == 0 {
> > -			return fmt.Errorf("router could not be created; you must specify a .kubeconfig file path containing credentials for connecting the router to the master with --credentials")
> > -		}
> > +	if err := validateServiceAccount(kClient, namespace, cfg.ServiceAccount, cfg.HostNetwork); err != nil {
> > +		return fmt.Errorf("router could not be created; %v", err)
>
> s/;/:/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7251/files#r53197443>.
>